ffi/util.lua

```diff
@@ -214,6 +214,23 @@ function util.execute(...)
     end
 end
 
+--- Returns the length of data that can be read without blocking
+function util.availableDataForReadingSize(luafile)
+    -- returns 0 if not readable yet, otherwise len of available data
+    -- returns -1 when unsupported: caller can read (with possible blocking)
```

Comment:
Is that better than plain old nil?

Reply:
Thought it would be more coherent to always return numbers.
`if util.availableDataForReadingSize(luafile) ~= 0 then go on`
It would mean getting nil would go on. But I don't know.

Reply:
That's no different for what you just wrote. However, it would allow you to safely write `if util.availableDataForReadingSize(luafile) > 0 then go on`

It might make more sense to write it like this though (in Lua, at least)

```lua
local something = util.availableDataForReadingSize(luafile)
if something and something > 0 then
```

Reply:
Hah, I just realized I this is exactly the same as https://github.com/koreader/koreader/pull/3231#discussion_r139309957

Reply:
Well, the most probable use case would be that if `util.availableDataForReadingSize()` is unsupported or not working and can't figure out if data is available, we would go on. So most probably `if something ~= 0`, which work with nil or -1. You tell me.
(So, I can assume you're fine with that long function name ? :)

Reply:
I see. So either way I see no need to deviate from nil then. :-P

I hadn't really thought about the function name yet. What about `util.safeReadingSize`?

Reply:
safe alone don't say much, safe_for_non_blocking_reading would :)
I let you think :)
(sorry for bothering with function and file names, but names itch me till i find a good one)

Reply:
or just util.availableReadSize(luafile)

Reply:
getSafeNonBlockReadingSize